src/control/server/mgmt_pool.go

```diff
@@ -503,23 +535,62 @@ func resolvePoolPropVal(req *mgmtpb.PoolSetPropReq) (*mgmtpb.PoolSetPropReq, err
 }
 
 // PoolSetProp forwards a request to the I/O server to set a pool property.
-func (svc *mgmtSvc) PoolSetProp(ctx context.Context, req *mgmtpb.PoolSetPropReq) (*mgmtpb.PoolSetPropResp, error) {
+func (svc *mgmtSvc) PoolSetProp(ctx context.Context, req *mgmtpb.PoolSetPropReq) (resp *mgmtpb.PoolSetPropResp, err error) {
 	svc.log.Debugf("MgmtSvc.PoolSetProp dispatch, req:%+v", req)
 
+	if err := svc.sysdb.CheckLeader(); err != nil {
+		return nil, err
+	}
+
 	newReq, err := resolvePoolPropVal(req)
 	if err != nil {
 		return nil, err
 	}
 
 	svc.log.Debugf("MgmtSvc.PoolSetProp dispatch, req (converted):%+v", newReq)
 
+	// Label is a special case, in that we need to ensure that it's unique
+	// and also to update the pool service entry.
+	if newReq.GetNumber() == drpc.PoolPropertyLabel {
+		uuidStr := newReq.GetUuid()
+		label := newReq.GetStrval()
+
+		uuid, err := uuid.Parse(uuidStr)
+		if err != nil {
+			return nil, errors.Wrapf(err, "failed to parse request uuid %q", uuidStr)
+		}
+		ps, err := svc.sysdb.FindPoolServiceByUUID(uuid)
+		if err != nil {
+			return nil, err
+		}
+
+		if label != "" {
+			// Setting new label, must be unique
+			_, err := svc.sysdb.FindPoolServiceByLabel(label)
+			if err == nil {
```

Comment:
don't we need to check that the error returned is not something other than Faultnot_found, what if the error is from CheckLeader?

Reply:
Yes, good catch.